sql-cli/noxfile.py

```diff
@@ -25,37 +25,42 @@ def dev(session: nox.Session) -> None:
 def test(session: nox.Session, airflow: str) -> None:
     """Run both unit and integration tests."""
 
+    session.virtualenv.env.update(
+        {
+            # Airflow automatically creates the ~/airflow directory on importing airflow
+            # but we are not using this directory hence we change the home to a tmp directory to avoid side effects.
+            "AIRFLOW_HOME": session.create_tmp(),
+        }
+    )
+
+    session.install("poetry")
+
     if airflow.startswith("2.2."):
         # To install some versions of Airflow, we need constraints, due to issues like:
         # https://github.com/apache/airflow/issues/19804
         constraints_url = (
             "https://raw.githubusercontent.com/apache/airflow/"
             f"constraints-{airflow}/constraints-{session.python}.txt"
         )
-
-        # Poetry does not support constraints:
-        # https://github.com/python-poetry/poetry/issues/3225
-        session.install("-e", ".", f"apache-airflow=={airflow}", "-c", constraints_url)
-
         # We are duplicating the tests dependencies until we find a better solution.
         # The solution might be to move out of poetry.
-        session.install("pytest")
-        session.install("pytest-cov")
-        session.install("mypy")
-        session.install("types-pyyaml")
-
-        session.log("Installed Dependencies:")
-        session.run("pip3", "freeze")
-
-        session.run("pytest", *session.posargs)
+        dev_deps = ("pytest", "pytest-cov", "mypy", "types-pyyaml")
+        # Poetry does not support constraints:
+        # https://github.com/python-poetry/poetry/issues/3225
+        session.run("poetry", "run", "pip", "install", "-e", ".", *dev_deps, "-c", constraints_url)
+        session.run("poetry", "run", "pip", "install", f"apache-airflow=={airflow}", "-c", constraints_url)
     else:
-        session.install("poetry")
         session.run("poetry", "install", "--with", "dev")
         session.run("poetry", "run", "pip", "install", f"apache-airflow=={airflow}")
 
-        session.log("Installed Dependencies:")
-        session.run("poetry", "run", "pip3", "freeze")
-        session.run("poetry", "run", "pytest", *session.posargs)
+    session.log("Installed Dependencies:")
+    session.run("poetry", "run", "pip", "freeze")
+
+    pytest_env = {
+        "TERMINAL_WIDTH": "3000",
+        "_TYPER_FORCE_DISABLE_TERMINAL": "1",
+    }
+    session.run("poetry", "run", "pytest", *session.posargs, env=pytest_env)
```

Comment:
I like this section ❤️ 

I was wondering if we should modify the make target call nox instead of poetry run here: https://github.com/astronomer/astro-sdk/blob/main/sql-cli/Makefile#L34

so that we're consistent with the CI run and local run make target.

Reply:
Yes, I also thought about that, but which python version and airflow version should we choose?

Reply:
Python 3.9 and Airflow 2.5 ?  :) 
I think runtime image uses Python 3.9 so and observed that my poetry somehow has Airflow 2.5

Reply:
poetry will always use the latest version possible specified in `pyproject.toml`.